### PR TITLE
Throw NLogConfigurationException if TimeToSleepBetweenBatches == 0

### LINKED ISSUE #1069: High CPU usage with async (with timeToSleepBetweenBatches="0")
I have the problem that the CPU has 20-30% usage all the time.
To reproduce the problem, you can simply create a project with just a Log statement and the following configurations:

    <nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" throwExceptions="true">
      <targets>
        
            <target name="Database" xsi:type="Database">
              <connectionStringName>XXX</connectionStringName>
              <commandText>XXX</commandText>
            </target>
        
      </targets>
      <rules>
        ....
      </rules>
    </nlog>

Someone already reports this bug a few years ago https://github.com/NLog/NLog/issues/162. Not sure that is was correctly fixed. I'm using version 4.2.1

### PR DESCRIPTION
**Pull request by vincechan**
fixes #1069.

### CODE REVIEW

**Comment by 304NotModified** [comment]
Thanks!

File: src/NLog/Targets/Wrappers/AsyncTargetWrapper.cs [hunk 1]
```diff
@@ -176,6 +176,11 @@ protected override void FlushAsync(AsyncContinuation asyncContinuation)
         /// </summary>
         protected override void InitializeTarget()
         {
+            if (this.TimeToSleepBetweenBatches == 0) {
+                throw new NLogConfigurationException(
+                    string.Format("The AysncTargetWrapper's TimeToSleepBetweenBatches property must be > 0"));
```
Comment (304NotModified) [inline src/NLog/Targets/Wrappers/AsyncTargetWrapper.cs]: why is this a string.format?

(please use commit amend and force push)

**Comment by vincechan** [comment]
string.Format removed.

**Comment by 304NotModified** [comment]
I guess you need to push it ;)

Is it also possible to squash/rebase the commits? 4 commits for less then 4 lines changed is a bit much

**Comment by 304NotModified** [comment]
Hi! 

Thanks for this PR. I will close it, but your commits are merged with the master. Did some small changes and rebase, in a new PR: https://github.com/NLog/NLog/pull/1336

Thanks for this submit! This will be released in 4.3 (RC2)

**Comment by vincechan** [comment]
Thank you.

**Pull request closed without merging**